### ISSUE TITLE
Detects incorrect git tag, and attempts to create tag that already exists

### ISSUE THREAD

**Issue by pathurs** [semantic-release/semantic-release#880]
## Current behavior

Previous gitTag is `v1.3.1`

## Expected behavior

Previous gitTag should be `v1.3.2`

## Environment

- **semantic-release** version: None (uses `npx semantic-release`)
- CI environment: Travis CI
- Plugins used: 
        "@semantic-release/changelog": "^3.0.0",
        "@semantic-release/git": "^7.0.1",
        "@semantic-release/github": "^5.0.1",
        "@semantic-release/npm": "^4.0.2",
- **semantic-release** configuration: https://github.com/Colonise/DataSource
- CI logs: https://travis-ci.org/Colonise/DataSource/builds/411787697

```
018-08-03T15:18:40.389Z semantic-release:get-last-release found tags: [ { gitTag: 'v1.3.2', version: '1.3.2' }, { gitTag: 'v1.3.1', version: '1.3.1' }, { gitTag: 'v1.3.0', version: '1.3.0' }, { gitTag: 'v1.2.0', version: '1.2.0' } ]
[3:18:40 PM] [semantic-release] › ℹ  Found git tag v1.3.1 associated with version 1.3.1
```

**Comment by pvdlg** [comment]
This happens because you already have a `v1.3.2` tag bu it's not in the `master` history.

```bash
$ git tag
// => v1.2.0
// => v1.3.0
// => v1.3.1
// => v1.3.2
```
```bash
$ git merge-base --is-ancestor v1.3.2 master
// => exit with 1
```

It seems you have rewritten your git history at some point, and now you have this `v1.3.2` that is referencing a commit that doesn't exist in any branch history.

So when looking at the history of `master` semantic-release find the last tag `v1.3.1`, increment the version and fails to create `v1.3.2` because it exists in the repo.

You should not create version like tags on branches other than your release branch.

**Comment by mrchief** [comment]
> This happens because you already have a v1.3.2 tag bu it's not in the master history.

Is there a way to recover from this?

**Comment by mrchief** [comment]
I was able to apply the workaround listed [here](https://github.com/semantic-release/semantic-release/blob/caribou/docs/support/troubleshooting.md#enotinhistory-commit-not-in-history). But say one cannot apply the listed workarounds for some reason - does that mean he/she cannot use SR anymore?

**Comment by pathurs** [comment]
It's a shame that semantic-release cannot detect this itself (yet?). Humans aren't perfect and will edit git history if needed (I do this every few months).

Here's how I've fixed this now (for tag: 'v1.2.3')

Get tag message (for later):
`git show v1.2.3`

Delete local tag:
`git tag -d v1.2.3`

Re-add local tag:
`git tag -a v1.2.3 -m <previous_tag_message>`

Tell origin about tag delete:
`git push origin :refs/tags/v1.2.3`

Tell origin about tag re-add:
`git push origin v1.2.3`
